src/main/java/makePost/use_case/MakePostDataAccessInterface.java

```diff
@@ -0,0 +1,22 @@
+package makePost.use_case;
+
+import entities.Post;
+
+import java.util.List;
+import java.util.Map;
+
+public interface MakePostDataAccessInterface {
+    int getNumberOfPosts();
+    void setNumberOfPosts(int newNumPostsCreated);
+    void savePost(Map<String, String> postAttributes);
+    int getCurrentUser();
+    List<Map<String, Object>> getFavouritedUsers();
```

Comment:
If we are reading from the database, does the map still have to map from strings to objects? Same thing for `getReplies`.